src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+                            return (Map<RandomGeneratorProperty, Object>)getProperties.invoke(null);
+                        };
+                        props = properties = AccessController.doPrivileged(getAction);
+                    } catch (SecurityException | NoSuchMethodException | PrivilegedActionException ex) {
+                        props = properties = Map.of();
+                    }
+                }
+            }
+        }
+
+        return props;
+    }
+
+    private Object getProperty(RandomGeneratorProperty property, Object defaultValue) {
+        return getProperties().getOrDefault(property, defaultValue);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private boolean isSubclass(Class<? extends RandomGenerator> category) {
+        return isSubclass(category, provider);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     * @param provider Provider that is being filtered.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private static boolean isSubclass(Class<? extends RandomGenerator> category,
+                                      Provider<? extends RandomGenerator> provider) {
+        return provider != null && category.isAssignableFrom(provider.type());
+    }
+
+    /**
+     * Returns the provider matching name and category.
+     *
+     * @param name      Name of RandomGenerator
+     * @param category  Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return A provider matching name and category.
+     *
+     * @throws IllegalArgumentException if provider is not a subclass of category.
+     */
+    private static Provider<? extends RandomGenerator> findProvider(String name,
+                                                                    Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        Map<String, Provider<? extends RandomGenerator>> fm = getFactoryMap();
+        Provider<? extends RandomGenerator> provider = fm.get(name.toUpperCase());
+        if (!isSubclass(category, provider)) {
+            throw new IllegalArgumentException(name + " is an unknown random number generator");
+        }
+        return provider;
+    }
+
+    /**
+     * Returns a stream of matching Providers.
+     *
+     * @param category  Sub-interface of {@link RandomGenerator} to type check
+     * @param <T>       Sub-interface of {@link RandomGenerator} to produce
+     *
+     * @return Stream of matching Providers.
+     */
+    static <T extends RandomGenerator> Stream<RandomGeneratorFactory<T>> all(Class<? extends RandomGenerator> category) {
```

Comment:
this signature is weird, T is not used in the parameter, so in case return any type of Stream<RandomGeneratorFactory<T>> from a type POV, should it be
` <T extends RandomGenerator> Stream<RandomGeneratorFactory<T>> all(Class<? extends T> category)` instead ?

Reply:
agree